### DISCUSSION TITLE
Support for websocket-connection-limits on a per-user basis

### DISCUSSION THREAD

**Discussion by dsidirop** [nats-io/nats-server#2802] [Q&A] [answered]
It's documented that NATS does support setting the websocket-connection-limit per Account. For example:

            server_name: S22
            max_connections: 1024
            
            no_auth_user: nats22

            accounts {
              default: {
                users = [ { user: nats22 } ]
                limits = {
                    max_connections: 256
            	max_payload: 4k
                }
              }
            }

Question: **How can we set the websocket-connection-limit per User (instead of per Account)?**

I'm asking this because in Synadia NGS managed service, even though it's true that from a technical standpoint there is no reason to have a limit to the number of accounts, practically it's not easy, at least not currently, to have thousands upon thousands of Accounts (typically a company only gets 2 or 3 accounts in NGS at the time of this writing due to billing constraints).

If a company uses NATS to run its drone-swarm that is comprised of 30k+ swarms then each one of these swarm will have its own user. And all these users live under just 1 Account.

Setting the connection-limit globally for the Account is nice for sure. But being able to fine-tune the connection-limits per user would really make the nats-server far more robust and alluring when it comes to industrial scenarios such as the aforementioned one.

Just my 2c.

**Comment by stevedischinger1** [comment, marked as answer]
There is not an artificial limit on the number of accounts in NGS. Also there are upcoming NGS updates that will allow multiple accounts under a single billing account.

**Reply by dsidirop** [reply]
NATS and Synadia NGS deliver! :)
